src/Features/Core/Diagnostics/BaseDiagnosticIncrementalAnalyzer.cs

```diff
@@ -19,24 +19,145 @@ protected BaseDiagnosticIncrementalAnalyzer(Workspace workspace)
         }
 
         #region IIncrementalAnalyzer
+        /// <summary>
+        /// Analyze a single document such that local diagnostics for that document become available,
+        /// prioritizing analyzing this document over analyzing the rest of the project.
+        /// Calls <see cref="DiagnosticAnalyzerService.RaiseDiagnosticsUpdated(object, DiagnosticsUpdatedArgs)"/> for each
+        /// unique group of diagnostics, where a group is identified by analysis classification (syntax/semantics), document, and analyzer.
+        /// </summary>
+        /// <param name="document"></param>
```

Comment:
If there are no param tags at all then VS won’t produce the error. It’s only when some of the params have tags that happens.

From: JohnHamby [mailto:notifications@github.com]
Sent: Friday, February 20, 2015 5:23 PM
To: dotnet/roslyn
Cc: Srivatsn Narayanan
Subject: Re: [roslyn] BaseDiagnosticIncrementalAnalyzer doc comments (#747)


In src/Features/Core/Diagnostics/BaseDiagnosticIncrementalAnalyzer.cs<https://github.com/dotnet/roslyn/pull/747#discussion_r25116906>:

> @@ -19,24 +19,145 @@ protected BaseDiagnosticIncrementalAnalyzer(Workspace workspace)

>          }

>

>          #region IIncrementalAnalyzer

> +        /// <summary>

> +        /// Analyze a single document such that local diagnostics for that document become available,

> +        /// prioritizing analyzing this document over analyzing the rest of the project.

> +        /// Calls <see cref="DiagnosticAnalyzerService.RaiseDiagnosticsUpdated(object, DiagnosticsUpdatedArgs)"/> for each

> +        /// unique group of diagnostics, where a group is identified by analysis classification (syntax/semantics), document, and analyzer.

> +        /// </summary>

> +        /// <param name="document"></param>

VS produces a warning, which turns into an error, if a parameter does not have a corresponding element.

I'll try to put some meaningful text in all the elements now--I wanted to draw attention to the points where my inferences might have been incorrect.

—
Reply to this email directly or view it on GitHub<https://github.com/dotnet/roslyn/pull/747/files#r25116906>.